### PR TITLE
Fix: distributionInfo miss sync for OSD type.

### PR DESCRIPTION
**Pull request by xuezhaojun**
Fixes: https://issues.redhat.com/browse/ACM-7888

Also need to backport to 2.8.3.

### CODE REVIEW

File: pkg/klusterlet/clusterinfo/distributionInfo_sync.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ type distributionInfoSyncer struct {
 func (s *distributionInfoSyncer) sync(ctx context.Context, clusterInfo *clusterv1beta1.ManagedClusterInfo) error {
 	// currently we only support OCP in DistributionInfo
 	switch clusterInfo.Status.KubeVendor {
-	case clusterv1beta1.KubeVendorOpenShift:
+	case clusterv1beta1.KubeVendorOpenShift, clusterv1beta1.KubeVendorOSD:
```
Comment (zhiweiyin318) [inline pkg/klusterlet/clusterinfo/distributionInfo_sync.go:28]: not only OCP and OSD, but also ARO, ROSA, ROKS which are in OCPList

Comment (xuezhaojun) [reply pkg/klusterlet/clusterinfo/distributionInfo_sync.go:28]: That's  a fun truth, in the `ProductOCPList`, other type will finally be represent by `KubeVendorOpenShift`: https://github.com/stolostron/multicloud-operators-foundation/blob/b4de56dec0b68530ad4701805403e2b66272e484/pkg/klusterlet/clusterinfo/defaultInfo_sync.go#L88

**Pull request merged**